### ISSUE TITLE
False positive error when overriding async generator

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7707]
This comes from an issue reported against mypy [here](https://github.com/python/mypy/issues/17128). Pyright produces the same false positive as mypy in this case.

Copying the sample code here for completeness.

```python
from abc import ABC, abstractmethod
from typing import AsyncIterator, Protocol


class BaseClass(ABC):
    @abstractmethod
    async def async_generator(self) -> AsyncIterator[int]:
        raise NotImplementedError


class SubClass(BaseClass):
    async def async_generator(self) -> AsyncIterator[int]:
        yield 1


class PBaseClass(Protocol):
    async def async_generator(self) -> AsyncIterator[int]: ...


class PSubClass(PBaseClass):
    async def async_generator(self) -> AsyncIterator[int]:
        yield 1
```

**Comment by JelleZijlstra** [comment]
This is not a false positive; it's explained in the mypy docs under https://mypy.readthedocs.io/en/stable/more_types.html#asynchronous-iterators.

**Comment by erictraut** [comment]
Ah yes, good point. Closing.
